AlarmPlus/Models/Schedule.swift

```diff
@@ -0,0 +1,187 @@
+        var timeString : String = ""
+        
+        if self.alarmTime[0] <= 12 {
+            timeString.append(String(self.alarmTime[0]))
+        } else {
+            timeString.append(String(self.alarmTime[0]-12))
+        }
+        
+        timeString.append(" : ")
+        
+        if self.alarmTime[1] > 9 {
+            timeString.append(String(self.alarmTime[1]))
+        } else {
+            timeString.append("0" + String(self.alarmTime[1]))
+        }
+        
+        if self.alarmTime[0] < 12 {
+            timeString.append(" am")
+        } else {
+            timeString.append(" pm")
+        }
+        
+        return timeString
+    }
+    
+    func addDateToActiveAlarms(_ day : Int) {
+        var dateComponents = DateComponents()
+        var calendar = Calendar(identifier: .gregorian)
+        let timeZone = TimeZone.current
+        calendar.timeZone = timeZone
+        
+        dateComponents.calendar = calendar
+        //set time for the new alarm
+        if alarmTime.count == 2 {
+            dateComponents = setHoursAndMinutesForNewAlarm(dateComponents)
+        } else {
+            print("Time is out of range")
+        }
+        
+        //set date for the new alarm
+        let date = getNextDate(dayOfTheWeek: day)
+        let nextScheduleDateComponents = calendar.dateComponents(in: .current, from: date)
+        dateComponents.day = nextScheduleDateComponents.day
+        dateComponents.month = nextScheduleDateComponents.month
+        dateComponents.year = nextScheduleDateComponents.year
+        
+        activeAlarms[day] = dateComponents.date
+
+    }
+    
+    //returns an int that represents what day of the week it is currently.
+    //Sunday = 1 and increases by one till Saturday = 7
+    //If a date can't be obtained will return 1 - Sunday
+    func getTodaysDayIndex() -> Int {
+        var todaysDayIndex = 0
+        let newDate = Date()
+        
+        let calendar = Calendar(identifier: .gregorian)
+        let todaysDateCompenents = calendar.dateComponents(in: .current, from: newDate)
+        
+        if let dayIndex = todaysDateCompenents.day {
+            todaysDayIndex = dayIndex
+        } else {
+            todaysDayIndex = 1
+        }
+        
+        return todaysDayIndex
+    }
+    
+    private func setHoursAndMinutesForNewAlarm(_ dateComponents: DateComponents) -> DateComponents{
+        var newDateComponents = dateComponents
+        
+        newDateComponents.hour = alarmTime[0]
+        newDateComponents.minute = alarmTime[1]
+        
+        return newDateComponents
+    }
+    
+    
+    private func getNextDate(dayOfTheWeek: Int) -> Date{
+        var newDate = Date()
+        
+        let calendar = Calendar(identifier: .gregorian)
+        let todaysDateCompenents = calendar.dateComponents(in: .current, from: newDate)
+        
+        if let todaysDayOfTheWeek = todaysDateCompenents.weekday {
+            if dayOfTheWeek == todaysDayOfTheWeek {
```

Comment:
You can use a guard statement here, so that way you immediately return from the function.

```
guard dayOfTheWeek != todaysDayOfTheWeek else {
     return newDate()
}
```